src/canvas/PathIntersection.tsx

```diff
@@ -58,22 +56,23 @@ export class PathIntersection {
             for (segmentIndex = 0; !intersectionPoint && segmentIndex < thatPathSegments.length; segmentIndex++) {
                 intersectionPoint = thatPathSegments[segmentIndex].isPointNearSegment(point, 10);
             }
-            
-            if (intersectionPoint?.equals(thatPath.getStart()) || intersectionPoint?.equals(thatPath.getEnd())) {
-                continue;
-            }
```

Comment:
I wasn't sure why we needed this check, I put a throw inside of it and tested a lot of stuff and it never threw. I think that since we snap to endpoints before calling this method, we don't need this check.

Reply:
I think you are right that we no longer need it. But this function shouldn't necessarily assume we have already snapped, since it doesn't know anything about snapping.

Reply:
I agree that talking about snapping on the doc here is not appropriate. But we could update the precondition to say something like "the point is not within 10 pixels of an endpoint of a path"

Reply:
or add the code back, I don't mind that much. What should I do?

Reply:
Now that I think about it more, maybe we should remove this check because the start and end point are along the pattern path. So the check doesn't really belong here any way.

Reply:
Yes, good thought, I agree. I moved the check out to _checkPathIntersectionAndSplit in the Renderer. This way we make extra sure we won't try to split a path too close to an endpoint.